include/dirent.h

```diff
@@ -22,67 +22,91 @@
 /* MP support header */
 #include "MacportsLegacySupport.h"
 
+/* Alter function names declared by <dirent.h> to get them out of the way */
+/* Note: These renamed names are non-functional */
+#if __MP_LEGACY_SUPPORT_FDOPENDIR__
+#define opendir __mpls_renamed_libc_opendir
+#define closedir __mpls_renamed_libc_closedir
+#define readdir __mpls_renamed_libc_readdir
+#define readdir_r __mpls_renamed_libc_readdir_r
+#define rewinddir __mpls_renamed_libc_rewinddir
+#define seekdir __mpls_renamed_libc_seekdir
+#define telldir __mpls_renamed_libc_telldir
+#endif
+
 /* Include the primary system dirent.h */
 #include_next <dirent.h>
 
+/* Remove the above macros to make way for the declarations below */
+#if __MP_LEGACY_SUPPORT_FDOPENDIR__
+#undef opendir
+#undef closedir
+#undef readdir
+#undef readdir_r
+#undef rewinddir
+#undef seekdir
+#undef telldir
+
+/* Fallback to __DARWIN_ALIAS if the other variants are not defined (?) */
+/* Note: I don't know if this makes sense */
```

Comment:
Just in case, you could do

```c
#ifdef __DARWIN_ALIAS_I
#  define __MPLS_ALIAS_I __DARWIN_ALIAS_I
#else
#  define __MPLS_ALIAS_I __DARWIN_ALIAS
#endif

#ifdef __DARWIN_INODE64
#  define __MPLS_INODE64 __DARWIN_INODE64
#else
#  define __MPLS_INODE64 __DARWIN_ALIAS
#endif
```

Reply:
Thanks. It's done.